src/components/icon/icon.spec.js

```diff
@@ -514,29 +510,39 @@ describe('MdIcon service', function() {
     });
 
     describe('icon group is not found', function() {
-      it('should log Error', function() {
-        var msg;
-        try {
-          $mdIcon('notfound:someIcon')
-            .catch(function(error) {
-              msg = error.data;
-            });
+      it('should log Error and reject', inject(function($log) {
+        var errorMessage = 'Cannot GET notfoundgroup.svg';
+        var caughtRejection = false;
+
+        $mdIcon('notfound:someIcon')
+          .catch(function(error) {
+            expect(error.data).toBe(errorMessage);
+            caughtRejection = true;
+          });
 
-          $httpBackend.flush();
-        } finally {
-          expect(msg).toEqual('Cannot GET notfoundgroup.svg');
-        }
-      });
+        $httpBackend.flush();
+
+        expect(caughtRejection).toBe(true);
+        expect($log.warn.logs[0]).toEqual([errorMessage]);
```

Comment:
In tests `$exceptionHandler` is overridden by `ngMock`. This test version has two modes `log` and `rethrow`. In both cases the exception is written to the `$exceptionHandler.errors` array.

Also in tests the `$log` is overridden by `ngMock`. This test version does not write to the console but instead writes to arrays, one for each of the possible logging methods, `log`, `warn`, `info`, `error`, `debug`.

In this particular test, I believe that we are not hitting `$exceptionHandler` at all as we are handling the error and calling `$log.warn`.